UKHomeOffice/nmsw-ui
# NMSW-1074 Add clientState: finalFal6 & set to false on FAL6 upload

# Ticket



----

## AC

New rules for when passenger is 'complete' on tasklist and CYA link is enabled

- IF declaration.clientState.passengers is no then show complete
- IF declaration.clientState.passengers is yes AND declaration.FAL6 is null then show required
- IF declaration.clientState.passengers is yes AND declaration.FAL6 is not null AND declaration.clientState.lastUpload is no then show in progress

----

## To test

Scenarios to test

- create a new declaration
- upload a FAL1
- continue to task list page
- > passenger status should be required

NO PASSENGERS

- click on passengers
- select no
- continue to task list page
- > passenger status should be complete

YES PASSENGERS > NO FAL 6

- click on passengers
- select yes
- continue
- leave page
- return to that declaration (task list page)
- > passenger status should be required

YES PASSENGERS > YES FAL 6 > NOT FINAL

- click on passengers
- select yes
- upload a valid FAL6
- continue
- click save and come back later (takes to task list page)
- > passenger status should be in progress


----

## Notes
